src/ophyd_async/core/mock_signal_utils.py

```diff
@@ -0,0 +1,183 @@
+import asyncio
+import itertools
+from contextlib import asynccontextmanager, contextmanager
+from typing import Any, Callable, Dict, Generator, Iterator, List
+from unittest.mock import ANY
+
+from ophyd_async.core.signal import Signal
+from ophyd_async.core.utils import ReadingValueCallback, T
+
+from .mock_signal_backend import MockSignalBackend
+
+
+def _get_mock_signal_backend(signal: Signal) -> MockSignalBackend:
+    assert isinstance(signal._backend, MockSignalBackend), (
+        "Expected to receive a `MockSignalBackend`, instead "
+        f" received {type(signal._backend)}. "
+    )
+    return signal._backend
+
+
+def set_mock_value(signal: Signal[T], value: T):
+    """Set the value of a signal that is in mock mode."""
+    backend = _get_mock_signal_backend(signal)
+    backend._set_value(value)
+
+
+def set_mock_put_proceeds(signal: Signal[T], proceeds: bool):
+    """Allow or block a put with wait=True from proceeding"""
+    backend = _get_mock_signal_backend(signal)
+
+    if proceeds:
+        backend.put_proceeds.set()
+    else:
+        backend.put_proceeds.clear()
+
+
+def set_mock_callback(signal: Signal[T], callback: ReadingValueCallback[T]) -> None:
+    """Monitor the value of a signal that is in mock mode"""
+
+    backend = _get_mock_signal_backend(signal)
+    return backend.set_callback(callback)
+
+
+# Gives the signal put wait_for a chance to complete
+_A_BIT = 1e-4
+
+
+@asynccontextmanager
+async def block_puts_during(*signals: List[Signal]):
+    for signal in signals:
+        set_mock_put_proceeds(signal, False)
+    yield
+    for signal in signals:
+        set_mock_put_proceeds(signal, True)
+
+    await asyncio.sleep(_A_BIT)
```

Comment:
Please can we get rid of this sleep and put it in the tests that need it instead, I'm happy with tests having a sleep in them, but not the core code